__tests__/app.test.js

```diff
@@ -75,33 +87,38 @@ describe("GET /api/articles", () => {
     });
   });
 
-  test("500: Internal Server Error", async () => {
-    const mockQuery = jest
-      .spyOn(db, "query")
-      .mockRejectedValueOnce(new Error("Database error"));
-    const response = await request(app).get("/api/articles").expect(500);
-    expect(response.body.msg).toBe("Internal Server Error");
-    mockQuery.mockRestore();
+  test("200: Returns articles filtered by topic", async () => {
+    const { body: response } = await request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200);
+    expect(response).toHaveLength(12);
   });
+  describe("Error handling for GET /api/articles", () => {
+    test("500: Internal Server Error", async () => {
+      await assertInternalServerError("/api/articles", "Internal Server Error");
+    });
 
-  test("400: Bad request, invalid sort_by", async () => {
-    const response = await request(app)
-      .get("/api/articles?sort_by=invalid")
-      .expect(400);
-    expect(response.body.msg).toBe("Invalid sort_by value: invalid");
-  });
-  test("400: Bad request, invalid order", async () => {
-    const response = await request(app)
-      .get("/api/articles?order=invalid")
-      .expect(400);
-    expect(response.body.msg).toBe("Invalid order value: invalid");
-  });
+    test("400: Bad request, invalid sort_by", async () => {
+      const response = await request(app)
+        .get("/api/articles?sort_by=invalid")
+        .expect(400);
+      expect(response.body.msg).toBe("Invalid sort_by value: invalid");
+    });
+    test("400: Bad request, invalid order", async () => {
+      const response = await request(app)
+        .get("/api/articles?order=invalid")
+        .expect(400);
+      expect(response.body.msg).toBe("Invalid order value: invalid");
+    });
 
-  test("400: Bad request, invalid order and sort_by", async () => {
-    const response = await request(app)
-      .get("/api/articles?order=invalid&sort_by=invalid")
-      .expect(400);
-    expect(response.body.msg).toBe("Invalid sort_by and order values");
+    test("400: Bad request, invalid order and sort_by", async () => {
+      const response = await request(app)
+        .get("/api/articles?order=invalid&sort_by=invalid")
+        .expect(400);
+      expect(response.body.msg).toBe(
+        "Invalid sort_by value: invalid and Invalid order value: invalid",
+      );
+    });
```

Comment:
More topic query tests needed:
200 - for a topic which exists but has no articles --> returns empty array (paper)